src/cmd/ksh93/sh/arith.c

```diff
@@ -194,24 +194,30 @@ static Namval_t *scope(Namval_t *np,struct lval *lvalue,int assign)
 	return np;
 }
 
-static Math_f sh_mathstdfun(const char *fname, size_t fsize, short * nargs)
+/* lookup a function in the standard math function table */
+static Math_f sh_mathstdfun(const char *fname, size_t fsize, short *nargs)
 {
 	const struct mathtab *tp;
-	char c = fname[0];
-	for(tp=shtab_math; *tp->fname; tp++)
+	char firstc = fname[0];
+	/* first byte of tp->fname is num. args and return type, unless empty */
+	for(tp=shtab_math; tp->fname[0]=='\0'; tp++)
```

Comment:
This change is incorrect; it inverts the sense of the second `for` expression, so that no math function is ever found. That `for` statement should not change.